src/plugins/intel_gpu/src/plugin/ops/condition.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "intel_gpu/plugin/program.hpp"
+#include "ngraph/op/if.hpp"
+#include "ie_ngraph_utils.hpp"
+
+#include "intel_gpu/primitives/condition.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+const size_t idx_true = 0;
+const size_t idx_false = 1;
+
+static cldnn::condition::branch gen_branch(Program& p, const std::shared_ptr<ngraph::op::v8::If>& op, size_t idx) {
+    cldnn::condition::branch branch;
+    const auto& internal_body = (idx == idx_true)? op->get_then_body() : op->get_else_body();
+
+    InferenceEngine::CNNNetwork body_network(internal_body);
+    {
+        // CNNNetwork change the input/output data type to fp32 when input/output data type is fp16
+        // To run internal body, rollback input/output data to original one.
+        size_t tidx = 0;
+        auto& model_inputs = internal_body->get_parameters();
+        for (auto& in : body_network.getInputsInfo()) {
+            auto input_data_type = InferenceEngine::details::convertPrecision(model_inputs[tidx++]->get_output_tensor(0).get_element_type());
+            if (in.second->getPrecision() != input_data_type)
+                in.second->setPrecision(input_data_type);
+        }
+
+        tidx = 0;
+        for (auto& out : body_network.getOutputsInfo()) {
+            const auto& model_output = internal_body->get_output_op(tidx++);
+            auto output_data_type = InferenceEngine::details::convertPrecision(model_output->get_output_tensor(0).get_element_type());
+            if (out.second->getPrecision() != output_data_type)
+                out.second->setPrecision(output_data_type);
+        }
+    }
+
+    auto config = p.get_config();
+    config.set_property(ov::intel_gpu::max_dynamic_batch(1));
```

Comment:
I've tried your branch, and it seems that there's an issue with allow_new_shape_infer() property for inner programs when one of the branches have single `Constant` node. The value for the flag is calculated in `Program::BuildProgram` so it doesn't depend on the outer program anyhow and may introduce mix of old and new shape infers

Reply:
And it seems that there's an issue with `propagate_constants` pass. I've got
```
[ ERROR ] Check '(mem.get_layout() == layout) || layout.is_dynamic()' failed at src/plugins/intel_gpu/src/graph/primitive_inst.cpp:144:
[GPU] Unexpected layout of input memory for parameter:Parameter_3879 node!
Node layout: f16:bfyx:1x4:nopad
Memory layout: f16:bfyx:2125x4:nopad
```
error on `propagate_constants` call for some mask-rcnn version with If primitive

Reply:
I succeed to run the model by fixing calc_output_layouts for single const inner network. thanks